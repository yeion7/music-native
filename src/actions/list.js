export const reciveItems = data => ({
  type: "RECIVE_ITEMS",
  payload: {
    albums: data.albums,
    tracks: data.tracks
  }
});

export const fetchReady = value => ({
  type: "FETCH_READY",
  value
});
