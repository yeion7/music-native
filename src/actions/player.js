import { fetchSongs } from "../utils/api";

export const onPressSong = song => {
  return {
    type: "CHANGE_PLAYING_SONG",
    song
  };
};

export const showPlayer = value => {
  return {
    type: "CHANGE_PLAYER_STATUS",
    value
  };
};

export const onPressAlbum = album => ({
  type: "CHANGE_SELECTED_PLAYLISTS",
  album
});

export const expandPlayer = value => ({
  type: "CHANGE_EXPAND_PLAYER",
  value
});

export function fecthAlbum(url) {
  return async (dispatch, getState) => {
    const songs = await fetchSongs(url);

    dispatch(onPressAlbum(songs));
    dispatch(showPlayer(true));
  };
}

export function playSong(song) {
  return async (dispatch, getState) => {
    dispatch(onPressSong(song));
    dispatch(showPlayer(true));
  };
}
