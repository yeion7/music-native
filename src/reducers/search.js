export function setItems(
  state = {
    text: "",
    items: {
      tracks: [],
      albums: []
    },
    fetchReady: false
  },
  action
) {
  switch (action.type) {
    case "SET_ITEMS":
      return { ...state, ...action.payload };
    default:
      return state;
  }
}
