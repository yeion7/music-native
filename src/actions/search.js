import { fetchItems } from "../utils/api";
import { reciveItems } from "../actions/list";

export const changeSearch = text => ({
  type: "TYPING",
  text
});

export function fecthList(text) {
  const thunk = async (dispatch, getState) => {
    dispatch(changeSearch(text));

    const data = await fetchItems(text);

    dispatch(reciveItems(data));
  };

  thunk.meta = {
    debounce: {
      time: 500,
      key: "RECIVE_ITEMS",
      cancel: false
    }
  };

  return thunk;
}
