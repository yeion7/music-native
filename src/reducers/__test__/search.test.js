import React from "react";
import "react-native";

import { setItems } from "../search";
import { onSetItems } from "../../actions/search";

describe("search reducer test ", () => {
  it("returns the same state on an unhandled action", () => {
    expect(setItems(null, { type: "_NULL_" })).toMatchSnapshot();
  });

  it("handles TYPING action", () => {
    expect(
      setItems(
        null,
        onSetItems({
          text: "aw",
          items: {
            tracks: [],
            albums: []
          },
          fetchReady: true
        })
      )
    ).toMatchSnapshot();
  });
});
