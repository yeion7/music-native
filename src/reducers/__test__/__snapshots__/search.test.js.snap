// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search reducer test  handles TYPING action 1`] = `
Object {
  "fetchReady": true,
  "items": Object {
    "albums": Array [],
    "tracks": Array [],
  },
  "text": "aw",
}
`;

exports[`search reducer test  returns the same state on an unhandled action 1`] = `null`;
